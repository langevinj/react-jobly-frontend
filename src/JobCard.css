.Apply{
    float: right;
    margin-right: 10px;
    margin-bottom: 10px;
}

.JobCard{
    width: 100%
}