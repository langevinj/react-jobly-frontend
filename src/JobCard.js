import React from 'react' 
import JoblyApi from './JoblyApi'
import './JobCard.css'
import { useLocalStorage } from './hooks'
import { useHistory } from 'react-router-dom'
import { renderJobButton } from './helpers'

function JobCard({ job }){
    const [user, setUser] = useLocalStorage("user")
    const history = useHistory();

    const applyForJob = async (evt) => {
        evt.preventDefault()
        let res = await JoblyApi.applyToJob(user.user.username, job.id)
        setUser(user => ({user: {...user.user, applications: [...user.user.applications, job.id]}}))
    }

    const unApply = async (evt) => {
        evt.preventDefault();
        let res = await JoblyApi.unapplyToJob(user.user.username, job.id)
        let oldApplications = [...user.user.applications]
        const idx = oldApplications.indexOf(job.id)
        if(idx > -1){
            oldApplications.splice(idx, 1);
        }
        setUser(user => ({ user: { ...user.user, applications: [...oldApplications ]}}))
        history.push("/jobs")
    }

    return(
        <div className="JobCard">
           <h6>{job.title}</h6>
           <p>Salary: {job.salary}</p> 
           <p>Equity: {job.equity}</p>
           {renderJobButton(user.user.applications, job.id, applyForJob, unApply)}
        </div>
    )
}

export default JobCard;