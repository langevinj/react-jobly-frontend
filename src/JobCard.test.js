import React from 'react'
import { render } from '@testing-library/react'
import JobCard from './JobCard'

it("renders without crashing", function() {
    render(<JobCard />)
});

it("matches snapshot", function() {
    let item = {title: "Tester", salary: 50000, equity: 0.5};
    const { asFragment } = render(<JobCard item={item} />)
    expect(asFragment()).toMatchSnapshot();
});