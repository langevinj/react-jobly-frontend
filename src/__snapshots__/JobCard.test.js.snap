// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobCard"
  >
    <h6 />
    <p>
      Salary: 
    </p>
    <p>
      Equity: 
    </p>
    <button
      class="Apply btn-primary rounded"
      data-toggle="button"
      type="button"
    >
      Apply
    </button>
  </div>
</DocumentFragment>
`;
